docroot/profiles/custom/sitenow/sitenow.install

```diff
@@ -1088,3 +1090,118 @@ function sitenow_update_8032() {
     throw new UpdateException('Query failed to execute.');
   }
 }
+
+/**
+ * Create banner blocks out of existing hero blocks.
+ */
+function sitenow_update_8033() {
+  $connection = Database::getConnection();
+
+  // Grab nids and section deltas for all nodes w/hero.
+  $query = $connection->select('node__layout_builder__layout', 'n');
+  $query->fields('n', ['entity_id', 'delta'])
+    ->condition('layout_builder__layout_section', '%uiowa_hero%', 'LIKE');
+  $results = $query->execute();
+
+  $block_storage = \Drupal::service('entity_type.manager')
+    ->getStorage('block_content');
+  $node_storage = \Drupal::service('entity_type.manager')
+    ->getStorage('node');
+
+  foreach ($results as $record) {
+    $nid = $record->entity_id;
+    $section_delta = $record->delta;
+
+    $node = $node_storage->load($nid);
+    $layout = $node->get('layout_builder__layout');
+    $section = $layout->get($section_delta)->getValue()['section'];
+
+    $section_arr = $section->toArray();
+    $components = $section->getComponents();
+    $new_components = [];
+
+    foreach ($components as $component) {
+      $component_arr = $component->toArray();
+      // Check that it is actually a hero.
+      if ($component_arr['configuration']['id'] === 'inline_block:uiowa_hero') {
+        $rev_id = $component_arr['configuration']['block_revision_id'];
+        $hero_block = $block_storage->loadRevision($rev_id);
+
+        // Continue only if there is no video involved.
+        if ($hero_block->get('field_uiowa_hero_video')->isEmpty()) {
+          // Grab field values.
+          $excerpt = $hero_block->get('field_uiowa_hero_excerpt')->getValue()[0];
+          $image = $hero_block->get('field_uiowa_hero_image')->getValue()[0];
+          $title = $hero_block->get('field_uiowa_hero_title')->getValue()[0];
+          $status = $hero_block->get('status')->getValue();
+          $link = NULL;
+
+          // Update excerpt format to minimal_html if filtered_html.
+          $excerpt['format'] = ($excerpt['format'] === 'filtered_html') ?
+            'minimal' : $excerpt['format'];
+
+          // Check if a link is present, and extract info.
+          if (preg_match('|(.*?)(<a.*?</a>)(.*?)|', $excerpt['value'], $matches)) {
+            $excerpt['value'] = $matches[1] . $matches[3];
+            preg_match('|<a.*?href="(.*?)".*?>(.*?)</a>|', $matches[2], $link_matches);
+            // Set URI based on link being absolute or relative.
+            $uri = (substr($link_matches[1], 0, 4) === 'http') ?
+              $link_matches[1] : \Drupal::request()->getSchemeAndHttpHost() . $link_matches[1];
+            $link = [
+              'uri' => $uri,
+              'title' => strip_tags($link_matches[2]),
+              'options' => [],
+            ];
+          }
```

Comment:
Just noticing this. I wanted to double check with @bspeare that this would be the desired outcome of this transition, to have a link pulled out of the hero excerpt and added explicitly as a link button on the new banner.

Reply:
The hero on https://coronavirus.uiowa.edu/ would be a good use case.   Would it remove that link markup and move it to the link field?

Reply:
As it's laid out here, it would @bspeare. If we allow for filtered_html in the banner, we can bring the content over directly, but to keep it limited to minimal will require some sort of fenagling like this.

Reply:
Thanks.   I definitely don't want to move to filtered_html :).

Reply:
As a reference, here's how the update_hooked coronavirus.uiowa.edu banner currently appears:
![image](https://user-images.githubusercontent.com/49074566/94297468-2b657480-ff2a-11ea-8251-3a675ef0fe89.png)

Reply:
@cory-skeers So, it seems like we'll need to detect whether the hero was using the "Hero: Left aligned" LB style and apply the appropriate banner style. Otherwise, we should maybe default to a center aligned (horizontally and vertically) variant?

Reply:
We probably have more banners that we are replacing that use the style that @cory-skeers has hooked up.   Perhaps we just manually update the covid home page?

Reply:
But we also don't want sites like https://americanstudies.prod.drupal.uiowa.edu/ to get this style.  Is there way to only add that style above to banners?

Reply:
I added a bit of structure to handle applying styles. The initial setup defaults to `banner_centered`, and then if the original hero image had `block_style_hero_left_aligned`, replaces it with `banner_centered_left`. We can change the default, or flesh it out with more logic if different handling options need to be applied.

Reply:
I'm cool with the approach. @bspeare and @mark-bennett-uiowa have a better sense of what the default should be so I'll defer to them.